recipe/meta.yaml

```diff
@@ -10,16 +10,18 @@ source:
   sha256: a14c1c5143081701d4fe62283e09cc5ad5336dff3a152de64eb77ee3180e9792
 
 build:
-  number: 1
+  number: 2
   noarch: python
 
 requirements:
   host:
     - python >=3.8
+    - dask-core {{ version }}
+    - distributed {{ version }}
   run:
     - python >=3.8
-    - dask-core ={{ version }},!={{ version }}a*
-    - distributed ={{ version }},!={{ version }}a*
+    - {{ pin_compatible('dask-core', max_pin='x.x.x') }}
+    - {{ pin_compatible('distributed', max_pin='x.x.x') }}
```

Comment:
Using the current version, this would generate the following constraints:

```
    - dask-core >=2022.8.0,<2022.8.1.0a0
    - distributed >=2022.8.0,<2022.8.1.0a0
```

One edge case @jakirkham noted that this wouldn't cover is nightly packages of security backports (i.e. something like `2022.8.0.1a20220819`; however, given the fact that it seems unlikely we would need to generate nightlies from backports, I think this is a reasonable solution.